examples/asr/didispeech-2/local/prepare_data.py

```diff
@@ -0,0 +1,179 @@
+
+    dic = dict(zip(key,value))
+    
+    files = []
+    is_sentid_line = True
+    for line in open(transcript_path, encoding="UTF-8-sig", errors='ignore'): #序号 文本 声韵母-类似标贝
+        if is_sentid_line:
+            transcript_id = line.split('\t')[0]
+            speaker = transcript_id.split('-')[0]
+            transcript = line.split('\t')[1].strip()
+            transcript = stringpartQ2B(transcript)#大写全角转小写半角
+            #去除符号
+            punc = '，。！？【】（）《》“‘：；［］｛｝&，．？（）＼％－＋￣~＄#＠=＿、／\\%-+~~$#@=_//,.!?' 
+            transcript = re.sub(r"[%s]+" %punc, "", transcript)
+            #print(transcript)           
+            wav_file = dic[str(transcript_id)]
+            wav_length = get_wave_file_length(wav_file)
+            files.append((os.path.abspath(wav_file), wav_length, speaker, transcript))
+        is_sentid_line = not is_sentid_line
+
+    df = pandas.DataFrame(
+            data=files, columns=["wav_filename", "wav_length_ms", "speaker", "transcript"])
+    df.to_csv(csv_path, index=False)
+    logging.info("Successfully generated total_csv file {}".format(csv_path)) ##output_dir/adult_total.csv    
+
+def split_train_dev_test(output_dir):
+    # get total_csv
+    total_csv_path = os.path.join(output_dir, 'total.csv')#output_dir/adult_total.csv
+    data = pandas.read_csv(total_csv_path)
+    x, y = data.iloc[:, :3], data.iloc[:, 3:]
+    # split train/dev/test---8:1:1
+    x_train, x_rest, y_train, y_rest = train_test_split(x, y, test_size=0.2, random_state=0)
+    x_test, x_dev, y_test, y_dev = train_test_split(x_rest, y_rest, test_size=0.5, random_state=0)
+    # add transcript
+    x_train.insert(2, 'transcript', y_train) #transcript 插入在speakr之前
+    x_test.insert(2, 'transcript', y_test)
+    x_dev.insert(2, 'transcript', y_dev)
+    # get csv_path
+    train_csv_path = os.path.join(output_dir, 'train.csv')
+    dev_csv_path = os.path.join(output_dir, 'dev.csv')
+    test_csv_path = os.path.join(output_dir, 'test.csv')
+    # generate csv
+    x_train.to_csv(train_csv_path, index=False, sep="\t")
+    logging.info("Successfully generated train_csv file {}".format(train_csv_path))
+    x_dev.to_csv(dev_csv_path, index=False, sep="\t")
+    logging.info("Successfully generated dev_csv file {}".format(dev_csv_path))
+    x_test.to_csv(test_csv_path, index=False, sep="\t")
+    logging.info("Successfully generated test_csv file {}".format(test_csv_path))
+
+def processor(dircetory):
+    """ download and process """   
+    haitian = os.path.join(dircetory, "didispeech-2.rar")
+    if os.path.exists(haitian):
+        logging.info("{} already exist".format(haitian))
+    else:
+        download_and_extract(dircetory, URL)
+        
+    convert_audio_and_split_transcript(dircetory)
+    split_train_dev_test(dircetory)
+    logging.info("Finished processing didispeech csv ")
+    
+if __name__ == "__main__":
+    logging.set_verbosity(logging.INFO)
+    INPUT_DIR = sys.argv[1] 
+    processor(INPUT_DIR)
```

Comment:
after we released download link for didispeech, please add download scripts here